framework/web_app/src/stores/productStore.ts

```diff
@@ -42,7 +42,9 @@ export const useProductStore = defineStore('product', () => {
      * @returns The new length of the stores array after adding the new merchants.
      */
     const addStoresToProductSearchFilter = (merchantsData: Merchant[]): number =>
-        productSearchOfferFilters.stores.push(...merchantsData);
+        productSearchOfferFilters.stores.push(
+            ...merchantsData.filter(merchant => merchant.is_enabled)
```

Comment:
This is just a general 'add' method.

IMO either:
- Filter the merchants argument to be filter by is enabled in the merhcantStore where this method is invoked
- OR re-purpose this method to not be a general add and rename it. E.g., addEnabledStoresToProductSearchFilter

Reply:
fixed :D using common method from store and moved this logic to there so this method is now dumb